components/RapDetails.js

```diff
@@ -56,12 +68,52 @@ export default function RapDetails() {
     setCartTickets([...cartTickets, ticketData]);
   }
 
+  async function handleAddReview(event) {
+    event.preventDefault(event.target);
+
+    const reviewId = () => {
+      return Math.random().toString(32).substring(2);
+    };
+
+    const formData = new FormData(event.target);
+    const reviewData = {
+      id: reviewId(),
+      name: formData.get("name"),
+      rating: Number(formData.get("rating")),
+      opinion: formData.get("opinion"),
+      artistName: data.name,
+    };
+
+    try {
+      if (!reviews) {
+        setReviews([reviewData]);
+      } else {
+        const existingReview = reviews.find(
+          (review) => review.id === reviewData.id
+        );
+        if (existingReview) {
+          console.log(`A review for '${data.name}' is already in the cart.`);
+        } else {
+          setReviews([...reviews, reviewData]);
+        }
+      }
+
+      await onSubmitReview(reviewData);
```

Comment:
I can't make out where this function `onSubmitReview` comes from. Currently, it's not coming from anywhere. You either need to import it and pass it down from a higher order component.